### ISSUE TITLE
CalDateTime equality operators handle null-values incorrectly.

### ISSUE THREAD

**Issue by minichma** [ical-org/ical.net#372]
`CalDateTime.operator ==` and `CalDateTime.operator !=` fail when comparing against null. This implies, that natural expressions like `((x == null) || (x <= y))` fail. This certainly violates the intended contract for == and !=. It probably should be implemented similar to:

```cs
public static bool operator ==(CalDateTime left, IDateTime right) => 
    (ReferenceEquals(left, null) || ReferenceEquals(right, null))
    ? ReferenceEquals(left, right):
    left.Equals(right);

public static bool operator !=(CalDateTime left, IDateTime right) =>
    (ReferenceEquals(left, null) || ReferenceEquals(right, null))
    ? !ReferenceEquals(left, right) :
    !left.Equals(right);
```

Other comparison operators should probably handle null values gracefully too. Here is, how [Nullable\<T\>](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/using-nullable-types) does it:
> When you perform comparisons with nullable types, if the value of one of the nullable types is null and the other is not, all comparisons evaluate to false except for != (not equal).

**Comment by rianjs** [comment]
Looks reasonable. I’ll merge a stylistically tweaked version of this this week.

**Comment by rianjs** [comment]
Available in nuget version 4.1.5:
https://www.nuget.org/packages/Ical.Net/4.1.5
